src/sidebar/components/TagListItem.js

```diff
@@ -0,0 +1,68 @@
+import { Icon, Link } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+
+/**
+ * @typedef TagListItemProps
+ * @prop {string|null} [href] - If present, tag will be linked to this URL
+ * @prop {(tag: string) => any} [onRemoveTag] - Callback for deleting a tag. If
```

Comment:
```suggestion
 * @prop {(tag: string) => void} [onRemoveTag] - Callback for deleting a tag. If
```

This component doesn't use the result of the callback, so `void` is more appropriate.